sdk/storage/Azure.Storage.Common/src/Shared/UriExtensions.cs

```diff
@@ -86,19 +87,11 @@ public static Uri AppendToPath(this Uri uri, string segment)
         /// <returns>Account name or null if not able to be parsed.</returns>
         public static string GetAccountNameFromDomain(string host, string serviceSubDomain)
         {
-            var accountEndIndex = host.IndexOf(".", StringComparison.InvariantCulture);
+            string subDomainPeriodAdded = '.' + serviceSubDomain + '.';
+            var accountEndIndex = host.IndexOf(subDomainPeriodAdded, StringComparison.InvariantCulture);
```

Comment:
how does this work with custom endpoints? Do we have tests covering that?

Reply:
For custom endpoints the logic is in the [UriBuilder Constructor logic](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs#L184), which gets called by our client constructors (example: [BlobBaseClient ](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs#L244)

We actually have existing tests that use a custom endpoint. https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs#L39